websockets/echo/src/server.rs

```diff
@@ -1,78 +1,111 @@
+use actix_ws::AggregatedMessage;
+use ractor::{ActorProcessingErr, ActorRef};
 use std::time::{Duration, Instant};
 
-use actix::prelude::*;
-use actix_web_actors::ws;
-
 /// How often heartbeat pings are sent
 const HEARTBEAT_INTERVAL: Duration = Duration::from_secs(5);
 
 /// How long before lack of client response causes a timeout
 const CLIENT_TIMEOUT: Duration = Duration::from_secs(10);
 
+#[derive(Debug)]
+pub(crate) enum AsMessage {
+    Ws(actix_ws::AggregatedMessage),
+    Hb,
+}
+
 /// websocket connection is long running connection, it easier
 /// to handle with an actor
-pub struct MyWebSocket {
-    /// Client must send ping at least once per 10 seconds (CLIENT_TIMEOUT),
-    /// otherwise we drop connection.
-    hb: Instant,
-}
+pub(crate) struct MyWebSocket;
 
 impl MyWebSocket {
-    pub fn new() -> Self {
-        Self { hb: Instant::now() }
+    async fn handle_hb(
```

Comment:
nit: do you need `&self` at all here? Since everything is mutated on the state, it seems like it would be "cleanest" to have a newtype for the state, and these are mutable fn's on the state.

Reply:
had a go but still need the zst for the actor impl block, so opted to keep these as-is